### ISSUE TITLE
2022 [CVPR] Thin-Plate Spline Motion Model for Image Animation

### ISSUE THREAD

**Issue by pomelyu** [pomelyu/paper-reading-notes#2]
## Introduction
An end-to-end approach for model transfer. Use thin-plate splines to estimate the optical flow and then use the multi-scale in-painting model to produce the final results.

[Github](https://github.com/yoyo-nb/Thin-Plate-Spline-Motion-Model)

## Method
<img width="1046" alt="image" src="https://user-images.githubusercontent.com/2679561/235576742-4ae4482e-6261-4a80-bd5c-ee9f2342e7ab.png">

- **Background motion predictor** predict the background affine transform($E_{bg}$)
- **Keypoint detector** detect KxN key points for each image. Create TPS transform from every N key points ($E_{kp}$)
- **Dense Motion Block** predict the optical flow and multi-scale occlusion mask from above-mentioned K+1 transform. Specifically, it predicts the weights of each transform and we can combine the transforms by softmax function
- **Inpaint Network** encode the source image and apply occlusion mask to every wrapped feature layers. the decoder part then produce the final results
<img width="467" alt="image" src="https://user-images.githubusercontent.com/2679561/235596353-52285b28-7252-41f3-b5db-25c54d887b9b.png">

#### Loss function
- Perceptual loss
- Equivariance loss to constraint the key points: $L_{eq} = | E_{kp}(\tau_{ran}(S)) - \tau_{ran}(E_{kp}(S) |$ for FOMM[^1] and MRAA[^2]
- Background loss: $L_{bg} = | A_{S \rightarrow T} * A_{T \rightarrow S} - I |$
- Inpainting loss: $L_{warp} = \sum | \tau(E_i(S)) - E_i(D) |$, $E_i$ is the i-th layer of inpainting network

## Highlight
- Dropout for TPS transformers: In training, randomly turn off one of the K TPS transform and assign the equal weight to the rest in order to prevent strapped in the local minimum
- The animation via disentanglement(AVD) model to apply the relative motion to better animate the target if the shape of source and target are not match. see MRAA[^2]

## Limitation
- Failed in the extreme cases, for example dog and giraffe

## Comments
- highly related: FOMM[^1] and MRAA[^2]

[^1]: [FOMM] First order motion model for image animation
[^2]: [MRAA] Motion representations for articulated animation - https://snap-research.github.io/articulated-animation
